server/pbenchinacan/load_keycloak.sh

```diff
@@ -1,110 +1,98 @@
+ADMIN_USERNAME="admin"
+ADMIN_PASSWORD="admin"
+
 echo
-echo "KEYCLOAK_HOST_PORT: $KEYCLOAK_HOST_PORT"
+echo "KEYCLOAK_HOST_PORT: ${KEYCLOAK_HOST_PORT}"
 
 echo
 echo "Getting admin access token"
 echo "--------------------------"
 
-ADMIN_TOKEN=$(curl -s -X POST "http://$KEYCLOAK_HOST_PORT/realms/master/protocol/openid-connect/token" \
+ADMIN_TOKEN=$(curl -s -X POST "http://${KEYCLOAK_HOST_PORT}/realms/master/protocol/openid-connect/token" \
   -H "Content-Type: application/x-www-form-urlencoded" \
-  -d "username=admin" \
-  -d 'password=admin' \
+  -d "username=${ADMIN_USERNAME}" \
+  -d "password=${ADMIN_PASSWORD}" \
   -d 'grant_type=password' \
   -d 'client_id=admin-cli' | jq -r '.access_token')
 
-echo "ADMIN_TOKEN=$ADMIN_TOKEN"
+echo "ADMIN_TOKEN=${ADMIN_TOKEN}"
 echo
 
 echo "Creating realm"
 echo "--------------"
 
-curl -i -X POST "http://$KEYCLOAK_HOST_PORT/admin/realms" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" \
+curl -i -X POST "http://${KEYCLOAK_HOST_PORT}/admin/realms" \
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" \
   -H "Content-Type: application/json" \
   -d '{"realm": "pbench", "enabled": true}'
 
 echo "Creating client"
 echo "---------------"
 
-CLIENT_CONF=$(curl -si -X POST "http://$KEYCLOAK_HOST_PORT/admin/realms/pbench/clients" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" \
+CLIENT_CONF=$(curl -si -X POST "http://${KEYCLOAK_HOST_PORT}/admin/realms/pbench/clients" \
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" \
   -H "Content-Type: application/json" \
-  -d '{"clientId": "pbench-client", "directAccessGrantsEnabled": true, "serviceAccountsEnabled": true, "redirectUris": ["http://localhost:8080/*"]}')
+  -d '{"clientId": "pbench-client", "directAccessGrantsEnabled": true, "serviceAccountsEnabled": true, "redirectUris": "'"${KEYCLOAK_REDIRECT_URI}"'"]}')
 
-CLIENT_ID=$(grep -o -e 'http://[^[:space:]]*' <<< $CLIENT_CONF | sed -e 's|.*/||')
-echo "client_id=$CLIENT_ID"
+CLIENT_ID=$(grep -o -e 'http://[^[:space:]]*' <<< ${CLIENT_CONF} | sed -e 's|.*/||')
+echo "client_id=${CLIENT_ID}"
 echo
 
 echo "Getting client secret"
 echo "---------------------"
 
-PBENCH_CLIENT_SECRET=$(curl -s -X POST "http://$KEYCLOAK_HOST_PORT/admin/realms/pbench/clients/$CLIENT_ID/client-secret" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" | jq -r '.value')
+PBENCH_CLIENT_SECRET=$(curl -s -X POST "http://${KEYCLOAK_HOST_PORT}/admin/realms/pbench/clients/${CLIENT_ID}/client-secret" \
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" | jq -r '.value')
 
-echo "PBENCH_CLIENT_SECRET=$PBENCH_CLIENT_SECRET"
+echo "PBENCH_CLIENT_SECRET=${PBENCH_CLIENT_SECRET}"
 echo
 
-echo "Creating client role"
+echo "Creating an 'ADMIN' role under pbench-server client of the pbench realm"
 echo "--------------------"
 
-curl -i -X POST "http://$KEYCLOAK_HOST_PORT/admin/realms/pbench/clients/$CLIENT_ID/roles" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" \
+curl -i -X POST "http://${KEYCLOAK_HOST_PORT}/admin/realms/pbench/clients/${CLIENT_ID}/roles" \
```

Comment:
I wonder if the realm name should be parameterized, as well.

And, you might as well use the variables where appropriate in line 66.